### PR TITLE
Closes #96 general issue increase test coverage for dataset_vignette.R

### LINKED ISSUE #96: General Issue: Increase test coverage for `dataset_vignette.R`
### Background Information

We need to increase test coverage for our functions. The below screenshot has the function and gap in test coverage that we need to write additional unit tests.

image

### Definition of Done

New test written that covers the red gap area for this function.

 You can get this report by running `devtools::test_coverage()`

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!
closes_issue96

### CODE REVIEW

File: tests/testthat/test-dataset_vignette.R [hunk 1]
```diff
@@ -1,23 +1,34 @@
 library(admiral.test)
 
-## Test 1: a 'knitr_kable' object is output when run outside pkgdown ----
-test_that("dataset_vignette Test 1: a 'knitr_kable' object is output when run outside pkgdown", {
+# dataset_vignette
+## Test 1: A 'knitr_kable' object is output when run outside pkgdown ----
+test_that("Test 1: A 'knitr_kable' object is output when run outside pkgdown", {
```
Comment (bms63) [inline tests/testthat/test-dataset_vignette.R:5]:
```suggestion
test_that("dataset_vignette Test 1: A 'knitr_kable' object is output when run outside pkgdown", {
```

We actually want it the other way.  :)  https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#set-up-the-test-script
image

File: tests/testthat/test-dataset_vignette.R [hunk 2]
```diff
@@ -1,23 +1,34 @@
 library(admiral.test)
```
Comment (bms63) [inline tests/testthat/test-dataset_vignette.R:1]: Can you remove this from the script

File: tests/testthat/test-dataset_vignette.R [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
   expect_s3_class(dataset_vignette(head(admiral_dm)), "knitr_kable")
 })
 
-## Test 2: a 'datatables' object is output when run inside pkgdown ----
-test_that("dataset_vignette Test 2: a 'datatables' object is output when run inside pkgdown", {
+## Test 2: A 'datatables' object is output when run inside pkgdown ----
+test_that("Test 2: A 'datatables' object is output when run inside pkgdown", {
   Sys.setenv(IN_PKGDOWN = "true")
   on.exit(Sys.setenv(IN_PKGDOWN = ""))
   expect_s3_class(dataset_vignette(head(admiral_dm)), "datatables")
```
Comment (bms63) [inline tests/testthat/test-dataset_vignette.R:13]: Can you also update the calls like this for test data?  Or just make some simple dummy data and remove the dependency all together.

```suggestion
  expect_s3_class(dataset_vignette(head(admiral.test::admiral_dm)), "datatables")
```

**Comment by bms63** [comment]
Thanks so much for doing your first PR.  Day 2!! :)

**Comment by thomas-neitmann** [comment]
Great to see your first PR @zdz2101! Sorry to be uber-nitpicky but in the future please use `_` rather than `-` in the branch name. Thanks!

File: tests/testthat/test-dataset_vignette.R (hunk 1 above)
Comment (bundfussr) [reply tests/testthat/test-dataset_vignette.R:5]: @zdz2101 , you can easily achieve the correct formatting by calling the "format test_that test file" addin (see https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#addin-admiraldevformat_test_that_file).

File: tests/testthat/test-dataset_vignette.R [hunk 4]
```diff
@@ -1,23 +1,48 @@
...
 
-## Test 3: a 'knitr_kable' object is output when run outside pkgdown ----
-test_that("dataset_vignette Test 3: a 'knitr_kable' object is output when run outside pkgdown", {
-  Sys.setenv(IN_PKGDOWN = "false")
-  on.exit(Sys.setenv(IN_PKGDOWN = ""))
-  expect_s3_class(
-    dataset_vignette(head(admiral_dm), display_vars = vars(STUDYID, USUBJID)),
-    "knitr_kable"
+## Test 3: An error is output when calling variable not in dataset ----
+test_that("dataset_vignette Test 3: An error is output when calling variable not in dataset", {
```
Comment (bms63) [inline tests/testthat/test-dataset_vignette.R:38]:
```suggestion
## Test 3: An error is outputed when calling variable not in dataset ----
test_that("dataset_vignette Test 3: An error is outputed when calling variable not in dataset", {
```

File: tests/testthat/test-dataset_vignette.R [hunk 5]
```diff
@@ -1,23 +1,48 @@
...
-test_that("dataset_vignette Test 1: a 'knitr_kable' object is output when run outside pkgdown", {
-  expect_s3_class(dataset_vignette(head(admiral_dm)), "knitr_kable")
+  expect_s3_class(dataset_vignette(dm), "knitr_kable")
+  expect_s3_class(dataset_vignette(dm, display_vars = vars(STUDYID, USUBJID)), "knitr_kable")
 })
 
-## Test 2: a 'datatables' object is output when run inside pkgdown ----
-test_that("dataset_vignette Test 2: a 'datatables' object is output when run inside pkgdown", {
+## Test 2: A 'datatables' object is output when run inside pkgdown ----
+test_that("dataset_vignette Test 2: A 'datatables' object is output when run inside pkgdown", {
```
Comment (bms63) [inline tests/testthat/test-dataset_vignette.R:20]:
```suggestion
## Test 2: A 'datatables' object is outputed when run inside pkgdown ----
test_that("dataset_vignette Test 2: A 'datatables' object is outputed when run inside pkgdown", {
```

File: tests/testthat/test-dataset_vignette.R [hunk 6]
```diff
@@ -1,23 +1,48 @@
-library(admiral.test)
+# dataset_vignette ----
+## Test 1: A 'knitr_kable' object is output when run outside pkgdown ----
+test_that("dataset_vignette Test 1: A 'knitr_kable' object is output when run outside pkgdown", {
```
Comment (bms63) [inline tests/testthat/test-dataset_vignette.R:3]:
```suggestion
## Test 1: A 'knitr_kable' object is outputed when run outside pkgdown ----
test_that("dataset_vignette Test 1: A 'knitr_kable' object is outputed when run outside pkgdown", {
```

**Review by bms63 — APPROVED** [review]
Looks good.  Just some grammar stuff - i was not sure how to spell outputted.

**Comment by bms63** [comment]
Also best practice is to put the issue #96 in the commit message

https://pharmaverse.github.io/admiraldev/devel/articles/git_usage.html#commits-from-the-terminal-in-rstudio

**Pull request merged**